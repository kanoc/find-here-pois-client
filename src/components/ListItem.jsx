import React from 'react';
import styled from 'styled-components';

const ListElement = styled.div`
    cursor: pointer;
    padding: 10px 18px 10px 24px;
    &:hover {
        background-color: #e9e9e9;
    }
    background: ${props => props.isSelected ? '#aaa' : '#fff' };
  }
`;

const ListItem = ({ item, isSelected, onItemClick }) => {

    const handleOnClick = () => {
        onItemClick(item.id)
    };

    return (
        <ListElement onClick={handleOnClick} isSelected={isSelected}>
            <div>{item.title}</div>
            <div><i>Rating: {item.rating}</i></div>
        </ListElement>
    )};

export default ListItem
