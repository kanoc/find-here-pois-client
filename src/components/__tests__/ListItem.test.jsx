import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from '../ListItem';

it('renders correctly', () => {
    const onItemClick = () => {};
    const item = {
        title: 'Foo',
        rating: 2
    };
    const tree = renderer
        .create(<ListItem item={item} isSelected={false} onItemClick={onItemClick}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
