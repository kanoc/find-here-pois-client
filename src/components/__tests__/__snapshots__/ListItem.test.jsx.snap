// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa hpfjMZ"
  onClick={[Function]}
>
  <div>
    Foo
  </div>
  <div>
    <i>
      Rating: 
      2
    </i>
  </div>
</div>
`;
